include/envoy/http/filter.h

```diff
@@ -36,6 +36,25 @@ enum class FilterHeadersStatus {
   // results in creating a header only request/response.
   // This status MUST NOT be returned by decodeHeaders() when end_stream is set to true.
   ContinueAndEndStream,
+  // Continue iteration to remaining filters, but do not end the stream.
+  //
+  // Used when a filter wants to add a body to headers-only request/response, but this body is not
+  // readily available.
+  // This causes the headers iteration to continue, but the stream does not end.
+  // The filter is responsible to continue the stream by providing a body through
```

Comment:
Currently, yes. We can support adding a body by calling `addEncodedBody`\ `addDecodedBody` then calling `continueEncoding`\`continueDecoding`, but Matt said there's no need to implement it for now. I also explained how it could be implemented in the TODO below.

I think calling `continueEncoding`\`continueDecoding` without adding a body should be an incorrect behavior as downstream filters will receive headers with `end_stream = false` then no body. This could cause some filters to do an incorrect behavior as they would assume there's an incoming body.  Right now I documented that the filter should reset the stream if it was unable to provide a body.

Reply:
Is it possible to add an ASSERT that prevents continue* from being called in this case? Or is there already one that will trigger?

Like you said I don't think there's a need to implement this but it would be great to have the current state of what is possible/not possible documented + ASSERTed on.

I think is another wording thing like my other comment: it would be good to explicitly call out that calling continue* won't work.

Thinking about this some more it might also be nice to point out that the filter probably doesn't have to explicitly reset, it can just rely on idle timeouts etc. to reset it.

Reply:
There's already an ASSERT which I added a message to make it a little bit clearer. I will update the documentation to make things more explicit (considering the other comment too). As for the reset part, I guess it's more future-proof and clearer to have the filter explicitly reset the stream, as in: "We failed to provide the body, this response failed, reset the stream" rather than "We failed to provide a body, do nothing".

Reply:
I updated the comments to be more explicit.